include/lo2s/process_controller.hpp

```diff
@@ -33,6 +33,52 @@ extern "C"
 
 namespace lo2s
 {
+/** A container mapping thread IDs to the PIDs of the processes a thread was
+ *  launched in.  In the language of the Linux kernel, this container maps
+ *  process identifiers (PIDs) to the corresponding task group IDs (TGIDs).
+ **/
+class ThreadGroups
+{
+public:
+    ThreadGroups() = default;
+
+    using tid_type = pid_t;
```

Comment:
I like the intent but stick to pid_t until we actually implement types for those. It will be easier to find all occurences, if they are not hidden behind typedefs.